main.cpp

```diff
@@ -70,28 +70,485 @@ write 3 UDTs below that EACH have:
+        int numberOfCUDACores;
+        //model name 
+        std::string modelName = "default GPU";
+        //price
+        float price = 100.00f;
+        //maximum SLI capability
+        int maxSLICapability = 2;
+
+        GraphicsAccelerator();
+        ~GraphicsAccelerator();
+        //accelerate the graphics
+        void accelerateGraphics();//attempts to run code in parallel
+        //set price of Graphics accelerator
+        bool setPrice(float newPrice = 100.0f);
+        //output cuda version and number of cores
+        std::string outputCUDAVersionAndCores();//returns a string representing the cuda capbility 
+
+        void boostTheGraphics(int toMultiply);
+        int parallelSpeedIncreaseFactor(int desiredFactor);
+        
+    };
+
+    //1) number of processor cores (int)
+    int numberOfProcessorCores;
+    //2) memory In GB (int)
+    int memoryInGB;
+    //3) graphics accelerator (std::string)
+    GraphicsAccelerator graphicsAccelerator;
+    //4) motherboard type (std::string)
+    std::string motherboardType;
+    //5) audio interface name (std::string)
+    std::string audioInterfaceName;
+
+    Computer();
+    ~Computer();
+    //1) run multiple processes 
+    void runMultipleProcesses();
+    //2) run memtest
+    void runMemtest();
+    //3) update Graphics Driver
+    //return true on connection success
+    bool updateGraphicsDriver(GraphicsAccelerator graphicsAccelerator);
+    //input the graphics accelerator to update drivers for
+    //returns true if driver updated successfully;
+
+    double analyzeEnergyConsumption(int numberOfSecondsPoweredOn);
+    std::string memoryTopologyBlocksPerCore();
+
+};
+Computer::Computer() : numberOfProcessorCores(5), memoryInGB(32), motherboardType("micro ATX"), audioInterfaceName("ableton")
+{
+    std::cout << "Computer being constructed" << std::endl; 
+}
+Computer::GraphicsAccelerator::GraphicsAccelerator() : isGSyncCapable(true), numberOfCUDACores(1200)
+{
+    std::cout << "GraphicsAccelerator being constructed" << std::endl;
+}
+void Computer::GraphicsAccelerator::accelerateGraphics()
+{
+    //do nothing
+}
+bool Computer::GraphicsAccelerator::setPrice([[maybe_unused]]float toPrice)
+{
+    return false;
+}
+std::string Computer::GraphicsAccelerator::outputCUDAVersionAndCores()
+{
+    return "hello world";
+}
+
+void Computer::runMultipleProcesses()
+{
+    //imagine running in parallel
+}
+void Computer::runMemtest()
+{    
+    std::cout << "memtest running on " << memoryInGB << "GB" << std::endl;
+}
+bool Computer::updateGraphicsDriver(GraphicsAccelerator gA)
+{
+    std::string throwAway = gA.outputCUDAVersionAndCores();  
+    return true;
+}
+
+Computer::~Computer()
+{
+    std::cout <<"Computer destructor" << std::endl;
+}
+Computer::GraphicsAccelerator::~GraphicsAccelerator()
+{
+    std::cout <<"GraphicsAccelerator destructor" << std::endl;
+}
+void Computer::GraphicsAccelerator::boostTheGraphics(int multiply)
+{
+    if(multiply >0)
```

Comment:
add spaces around `<<` and `>` so the code is easier to read.
Formatting matters!!
- [ ] revise